### ISSUE TITLE
Return to user's last seen page on PaginatedState after timeout

### ISSUE THREAD

**Issue by gsvr** [praekelt/vumi-jssandbox-toolkit#223]
It would be ideal if this would also work if a special timeout state is interjected like in go-refugeerights `state_timed_out`.

**Comment by hodgestar** [comment]
I think this could be approached in one of two ways. Either we could add an option for setting and retrieving ``page_start`` (i.e. the current starting point for the list) or you could arrange for the go-refugeerights time out state to record the previous state's metadata in addition to its ``name`` and ``creator_opts``. The latter solution covers all state types and so is probably better for go-refugeerights. Having the option of setting ``page_start`` might still be nice though.

**Comment by gsvr** [comment]
Yeah, I thought it might be two different implementations for with and without a `timed_out` state, but thought I'd leave that comment in case it could be architected to work for both. I think it would still be useful in the cases where we don't have a `timed_out` state.
